discovery-client/src/main/java/io/micronaut/discovery/spring/config/SpringCloudConfigurationClient.java

```diff
@@ -89,7 +89,8 @@ protected SpringCloudConfigurationClient(SpringCloudConfigClient springCloudConf
         if (!configuredApplicationName.isPresent()) {
             return Flowable.empty();
         } else {
-            String applicationName = configuredApplicationName.get();
+            String applicationName = springCloudConfiguration.getName() == null ?
+                    configuredApplicationName.get() : springCloudConfiguration.getName();
```

Comment:
This is a minor thing but `springCloudConfiguration.getName()` could be extracted to a local variable so you won't have to invoke that method twice.

Reply:
Ok, I am going to refactor it.

Reply:
@jeffbrown thanks for your review, the code was updated.